### DISCUSSION TITLE
Incorporating PoS information explicitly for the `edit_tree_lemmatizer`

### DISCUSSION THREAD

**Discussion by oroszgy** [explosion/spaCy#10108] [Help: Best practices]
Hey, 
we are using the new experimental edit tree lemmatizer for Hungarian in [huspacy](https://github.com/huspacy/huspacy), and we are looking for a way to explicitly incorporate PoS information for the model, if it is possible at all. As far as I understood multitask learning is one way to help the model yielding PoS aware predictions. Is there any other way to condition the lemmatizer's predictions on PoS tags? For instance, is it possible to learn embeddings for part-of-speech tags in `MultiHashEmbed`? If yes, could you please guide me how to modify [this config](https://github.com/huspacy/huspacy/blob/experimental-edit-tree-lemmatizer/hu_core_news_lg/configs/edit-tree-lemmatizer.cfg#L29)?

**Comment by oroszgy** [comment]
@kadarakos - this is the problem I mentioned

**Comment by adrianeboyd** [comment]
You need to use "annotating components" and include the feature in the `attrs`. Docs: https://spacy.io/usage/training#annotating-components

Here's a demo project that includes `DEP` as a tagger feature:

https://github.com/explosion/projects/tree/v3/pipelines/tagger_parser_predicted_annotations

**Comment by oroszgy** [comment]
Wow, that’s awesome! Thanks! :)

**Reply by adrianeboyd** [reply]
As a note, I wouldn't be surprised if this didn't help, since the tagger's `tok2vec` already includes all the same underlying features, so adding a predicted (not gold) tag isn't adding much new info.

**Reply by honnibal** [reply]
Yeah the POS tagger doesn't even use any structured prediction tricks -- it's just one softmax over all the token vectors. So I wouldn't expect it to help. The parser features have a better chance. You could try putting the lemmatizer at the end of the pipeline and use the `DEP` feature perhaps.

**Comment by KennethEnevoldsen** [comment]
Hi @oroszgy, just wanted to let you know that the OdyCy pipeline (see paper [here](https://aclanthology.org/2023.latechclfl-1.14.pdf)) performs a multistep approach for lemmatisation that uses a heuristic fallback from the lookup to the learned lemmatizer. From my understanding, it is good if you have a lexicon with high precision but low recall.

**Reply by oroszgy** [reply]
Thanks for the update. We ended up doing sg. very similar for [HuSpaCy](https://github.com/huspacy/articles/releases/download/TSD_2023/HuSpaCy___TSD_2023.pdf).

**Reply by KennethEnevoldsen** [reply]
Oh, that looks interesting. Thanks for sharing it will def. have a look.
